graphs/possible_bipartition.py

```diff
@@ -5,8 +5,51 @@ def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(N + E)
+        Space Complexity: O(N)
     """
-    pass
+
+    if not dislikes:
+        return True
+
+    queue = deque()
+    dogs = set()
+    group_1 = set()
+    group_2 = set()
+
+    if dislikes[0]:
+        start = 0
+    else:
+        start = 1
```

Comment:
Consider an edge case where `dislikes` has multiple nodes at the beginning of the list that are disconnected from the graph. How might your refactor your code to address this edge case?